quill-core/src/main/scala/io/getquill/norm/RenameProperties.scala

```diff
@@ -98,28 +103,29 @@ object RenameProperties extends StatelessTransformer {
             (Join(typ, a, b, iA, iB, onr), Tuple(List(schemaA, schemaB)))
         }
 
-      case FlatJoin(typ, a: Query, iA, on) =>
+      case FlatJoin(typ, a: Ast, iA, on) =>
         applySchema(a) match {
           case (a, schemaA) =>
             val replaceA = replacements(iA, schemaA)
             val onr = BetaReduction(on, replaceA: _*)
             (FlatJoin(typ, a, iA, onr), schemaA)
         }
 
-      case Map(q: Operation, x, p) if x == p =>
-        (Map(apply(q), x, p), Tuple(List.empty))
+      case Infix(parts, List(param)) =>
+        val (p, schema) = applySchema(param)
+        (Infix(parts, List(p)), schema)
```

Comment:
Amount all of the changes only these 3 lines are actual fix, is it correct?
If this is true, you don't need to change the definition of `applySchema` from Query to Ast, since this breaks the original idea afaik (plus this introduce the user unfriendly exception).
As mentioned before, you can use approach as in https://github.com/getquill/quill/pull/1134/files#diff-1fd78b0579cdd5318ca84f4c836dd991R15. But since there's no `apply(Infix):Infix` in a `StatelessTransformer`, you can override the `apply(Ast):Ast` and then body would look like:
```scala
case Infix(..) => infix fix logic
case other => super.apply(other)
```
Such pattern is commonly used among the implementations of `StatelessTransformer`

Reply:
Oh sorry, now I see why this is needed. Also matching only single ast in tuple could be acceptable if we need a fix sine don't have a workaround. But first, I'll check it myself

Reply:
👍 thanks a lot ;-)